presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1193,12 +1197,16 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             }
             analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
 
+            long startTime = System.nanoTime();
             Optional<ViewDefinition> optionalView = metadata.getView(session, name);
+            session.getRuntimeStats().addMetricValue(GET_VIEW_TIME_NANOS, System.nanoTime() - startTime);
```

Comment:
Instead of having a lot of `System.nanoTime()` and `System.nanoTime() - startTime`, have we considered something like having the following methods in `RuntimeStats`?

```java
public <V> V profile(String tag, Supplier<V> supplier)
{
    long startTime = System.nanoTime();
    V result = supplier.get();
    addMetricValue(tag, System.nanoTime() - startTime);
    return result;
}

public void profile(String tag, Runnable runnable)
{
    profile(tag, () -> {
        runnable.run();
        return null;
    });
}
```

Then we can have
```java
Optional<ViewDefinition> optionalView = session.getRuntimeStats().profile(GET_VIEW_TIME_NANOS, () -> metadata.getView(session, name));
```